### PR TITLE
Add api definitions for transaction request

### PR DESCRIPTION
**Pull request by ghazninattarshah**

### CODE REVIEW

File: v1/transactionrequest/transactionrequest.gunk [hunk 1]
```diff
@@ -0,0 +1,660 @@
...
+type DebtorAccount struct {
+	Iban string `pb:"1" json:"iban"`
+}
+
+type CreditorAccount struct {
+	Iban string `pb:"1" json:"iban"`
+}
+
+type ToSepaCreditTransfers struct {
+	DebtorAccount    DebtorAccount   `pb:"1" json:"debtorAccount"`
```
Comment (nkristianto) [inline v1/transactionrequest/transactionrequest.gunk]: I think this json tag should be in snake_case

File: v1/transactionrequest/transactionrequest.gunk [hunk 2]
```diff
@@ -0,0 +1,660 @@
...
+	Description            string       `pb:"4" json:"description"`
+	Posted                 time.Time    `pb:"5" json:"posted"`
+	Completed              time.Time    `pb:"6" json:"completed"`
+	TransactionRequestType string       `pb:"7" json:"transaction_request_type"`
+	ChargePolicy           string       `pb:"8" json:"charge_policy"`
+}
+
+// TransactionRequestService provides transaction request related operations.
+type TransactionRequestService interface {
+
```
Comment (nkristianto) [inline v1/transactionrequest/transactionrequest.gunk]: Remove this line maybe

File: v1/transactionrequest/transactionrequest.gunk [hunk 3]
```diff
@@ -0,0 +1,660 @@
...
+	// +gunk openapiv2.Operation{
+	//         Tags:        []string{"TransactionRequest"},
+	//         Description: "Answer the transaction request challenge",
+	//         Summary:     "Answer the transaction reqeust challenge",
+	//         Responses: map[string]openapiv2.Response{
+	//                 "201": openapiv2.Response{
+	//                         Description: "Answered the transaction request challenge",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/transactionrequest",
```
Comment (nkristianto) [inline v1/transactionrequest/transactionrequest.gunk]: Wrong json tag

File: v1/transactionrequest/transactionrequest.gunk [hunk 4]
```diff
@@ -0,0 +1,659 @@
...
+	//                 },
+	//         },
+	// }
+	AnswerTransactionRequestChallenge(AnswerTransactionRequestChallengeRequest) AnswerTransactionRequestChallengeResponse
+
+	// CreateAccountTransaction creates a new account transaction request
+	//
+	// +gunk http.Match{
+	//         Method: "POST",
+	//         Path:   "/v1/banks/{BankID}/accounts/{AccountID}/transaction-request-types/ACCOUNT/transactionrequest",
```
Comment (nkristianto) [inline v1/transactionrequest/transactionrequest.gunk:343]: should we use the uppercase for ACCOUNT?

Comment (ghazninattarshah) [reply v1/transactionrequest/transactionrequest.gunk:343]: It was like that in the API to differentiate the types ACCCOUNT, ACCOUNT_OTP, SEPA, COUNTERPARTY, and FREE_FORM. I think we should leave it that way. @nkristianto

**Review by nkristianto — APPROVED** [review]
LGTM, feel free to merge it

**Pull request merged**